webplugin/js/app/kommunicate-ui.js

```diff
@@ -589,15 +589,33 @@ handleAttachmentIconVisibility : function(enableAttachment, msg, groupReloaded)
             hasMultipleConversations ? backButton.classList.remove('force-n-vis') : backButton.classList.add('force-n-vis')
         }
     },
-    handleWaitingQueueMessage: function (data) {
-        let waitingStatus = data && data.metadata.CONVERSATION_STATUS ==Kommunicate.conversationHelper.status.WAITING;
-        if (waitingStatus) {
-            kommunicateCommons.modifyClassList({id:["mck-waiting-queue"]}, "vis","n-vis");
-           
-        } else {
-            kommunicateCommons.modifyClassList({id:["mck-waiting-queue"]}, "n-vis","vis");
-        }
-    },
+        handleWaitingQueueMessage: function (data) {
+            let groupId = data && data.clientGroupId;
+            let waitingStatus = data && data.metadata.CONVERSATION_STATUS == Kommunicate.conversationHelper.status.WAITING;
+            mckUtils.ajax({
+                type: 'GET',
+                url: MCK_BASE_URL + '/rest/ws/group/waiting/list',
+                global: false,
+                contentType: 'application/json',
+                success: function (res) {
+                    WAITING_QUEUE = res.response;
+                    document.getElementById('mck-waiting-queue-2') && (document.getElementById('mck-waiting-queue-2').innerHTML = "#" + parseInt(WAITING_QUEUE.indexOf(parseInt(groupId)) + 1));
```

Comment:
Check if the status code or message "success" is available in the API to verify if results are properly fetched or not.